marklogic/src/main/scala/quasar/physical/marklogic/fs/package.scala

```diff
@@ -16,24 +16,59 @@
 
 package quasar.physical.marklogic
 
+import quasar.Predef._
+import quasar.Data
+import quasar.effect.{KeyValueStore, MonotonicSeq, Read}
+import quasar.fp._
+import quasar.fp.free._
 import quasar.fs._
-import quasar.fs.mount.FileSystemDef, FileSystemDef.DefErrT
+import quasar.fs.mount.{ConnectionUri, FileSystemDef}, FileSystemDef.DefErrT
+
+import com.marklogic.client._
+import com.marklogic.xcc._
+import java.net.URI
 
 import scalaz._, Scalaz._
 import scalaz.concurrent.Task
+import scalaz.stream.Process
 
 package object fs {
   val FsType = FileSystemType("marklogic")
 
-  def definition[S[_]](implicit S0: Task :<: S, S1: PhysErr :<: S)
-    : FileSystemDef[Free[S, ?]] = FileSystemDef.fromPF {
+  type Eff0[A] = Coproduct[
+                    KeyValueStore[ReadFile.ReadHandle, Process[Task, Vector[Data]], ?],
+                    KeyValueStore[WriteFile.WriteHandle, Unit, ?],
+                    A]
+  type Eff1[A] = Coproduct[MonotonicSeq, Eff0, A]
+  type Eff2[A] = Coproduct[Read[Client, ?], Eff1, A]
+  type Eff[A]  = Coproduct[Task, Eff2, A]
+
+  def inter(uri: ConnectionUri): Task[Eff ~> Task] = {
```

Comment:
Then again, unclear if we will need `DatabaseClient` at all. Which is why I am moving ahead with `it` testing before polishing this code.